java/impl/aws/src/main/java/co/worklytics/psoxy/aws/ParameterStoreConfigService.java

```diff
@@ -171,39 +177,48 @@ public boolean acquire(@NonNull String lockId, @NonNull Duration expires) {
 
         final String lockParameterName = lockParameterName(lockId);
         try {
-            client.putParameter(PutParameterRequest.builder()
+            Instant instant = clock.instant();
+            String lockValue = String.format("locked_%d", instant.toEpochMilli());
+            PutParameterResponse locked = client.putParameter(PutParameterRequest.builder()
                 .name(lockParameterName)
                 .type(ParameterType.STRING)
-                .value("locked")
+                .value(lockValue)
                 .overwrite(false)
                 .build());
-            return true;
+            // assuming two threads at the same time trying to write on a deleted parameter might
+            // not get ParameterAlreadyExistsException and last one wins.
+            Uninterruptibles.sleepUninterruptibly(100 + randomNumberGenerator.nextInt(200), TimeUnit.MILLISECONDS);
+            // if value doesn't match what we wrote, someone else got it, race condition.
+            Parameter parameter = readParameter(lockParameterName);
+            return Objects.equals(lockValue, parameter.value());
         } catch (ParameterAlreadyExistsException e) {
             try {
-                Instant lockedAt = client.getParameter(GetParameterRequest.builder()
-                    .name(lockParameterName)
-                    .build()).parameter().lastModifiedDate();
-
+                Instant lockedAt = readParameter(lockParameterName).lastModifiedDate();
 
                 if (lockedAt.isBefore(clock.instant().minusSeconds(expires.getSeconds()))) {
                     log.warning("Lock " + lockParameterName + " is stale, removing");
 
-                    //q: add random delay here, in case multiple instances have been waiting to
-                    // acquire the lock?
+                    // release it and let the caller decide if the retry
                     release(lockId);
-
-                    return acquire(lockId);
```

Comment:
removed this, so now will wait in caller WAIT_AFTER_FAILED_LOCK_ATTEMPTS + buffer